allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_data_transforms.py

```diff
@@ -266,15 +266,8 @@ def get_trials(self) -> pd.DataFrame:
             A dataframe containing behavioral trial start/stop times,
             and trial data
         """
-        timestamps = self.get_stimulus_timestamps()
-        licks = self.get_licks()
-        data = self._behavior_stimulus_file()
-        rewards = self.get_rewards()
 
-        trial_df = get_trials_from_data_transform(data,
-                                                  licks,
-                                                  rewards,
-                                                  timestamps)
```

Comment:
Not sure how I feel about this change. On the one hand it simplifies the call which is good. But it makes figuring out what exact data is needed for creating a trials table rather opaque.

Reply:
My thinking was:

All of the data that `get_trials_from_data_transform` actually needed (lick times, reward times, monitor delay) comes from a coherent data set. Allowing users to pass in a licks_df, a rewards_df, a stimulus_df, and a timestamps array by hand was just asking them to pass in something that was inconsistent. By passing in a single instance of `BehaviorDataTransforms` or `BehaviorOphysDataTransforms`, we can guarantee that the data is as consistent as the transformer class is.

Would it alleviate your fears if I added some checks at the beginning of `get_trials_from_data_transform` to the effect that, for instance, if the input data transform does not have `get_monitor_delay` it will raise an error with a message explicitly stating the problem.

Something like
```
missing_data_streams = []
if not hasattr(input_obj, 'get_monitor_delay'):
    missing_data_streams.append('monitor delay')
if not hasattr(input_obj, 'get_stimulus_timestamps'):
    missing_data_streams.append('stimulus time stamps')
...

if len(missing_data_streams) > 0:
    raise ValueError(f"could not calculate trials dataframe; missing {missing_data_streams}")
```

?

Reply:
Yeah, I think keeping the old documentation describing stuff like `data`, `licks`, etc.. would also help

Reply:
Cool. I can add that (probably won't get pushed up until Monday, since it involves new unit tests to make sure the right errors get thrown under the right circumstances).

Reply:
Done